civis/service_client.py

```diff
@@ -0,0 +1,143 @@
+import civis
+from collections import OrderedDict
+import re
+import six
+import warnings
+
+from jsonref import JsonRef
+import requests
+
+from civis.base import Endpoint, CivisAPIError, CivisAPIKeyError, tostr_urljoin
+from civis.resources._resources import parse_method
+from civis._utils import to_camelcase
+
+
+def auth_service_session(session, service_id):
+    try:
+        service = civis.APIClient().services.get(service_id)
+    except CivisAPIError as err:
+        if err.status_code == 404:
+                    msg = 'There was an issue connecting to your service!'
+                    six.raise_from(ValueError(msg), err)
+        else:
+            raise
```

Comment:
I wasn't sure about the error handling either. I figured it might be confusing to pass along the CivisAPIError. because to the user they are not interacting with the CivisAPI, they are interacting with ToolboxClient. The ToolboxClient is using CivisAPI to get their service information, which they may or may not know about.

I copied the syntax from [here](https://github.com/ggarcia-civis/civis-python/blob/945ca4c94aee5200ba7f28145556bec30e0e46fa/civis/io/_files.py#L407).